src/rf/apps/workers/thumbnail.py

```diff
@@ -118,13 +120,18 @@ def make_thumbs_for_layer(layer_id):
     for image in layer_images:
         thumb_dims = [IMAGE_THUMB_SMALL_DIMS, IMAGE_THUMB_LARGE_DIMS]
         try:
+            image = status_updates.update_layerimage_status_start(
+                image, enums.STATUS_THUMBNAIL)
             image.thumb_small_key, image.thumb_large_key = \
                 s3_make_thumbs(image.get_s3_key(), user_id,
                                thumb_dims, THUMB_EXT)
         except ImageCouldNotOpenError:
+            image = status_updates.update_layerimage_status_end(
+                image, enums.STATUS_THUMBNAIL, ERROR_MESSAGE_THUMBNAIL_FAILED)
             return False
 
-        image.status = enums.STATUS_THUMBNAILED
+        image = status_updates.update_layerimage_status_end(
+            image, enums.STATUS_THUMBNAIL)
```

Comment:
Consider moving `status_updates` functions to instance methods. 

`image.update_status_end(...)` is more concise than `status_updates.update_layerimage_status_end(image, ...)`.